editor/src/clj/editor/gui.clj

```diff
@@ -1020,16 +1022,15 @@
 (g/defnode ShapeNode
   (inherits VisualNode)
 
-  (property manual-size types/Vec3 (default (protobuf/vector4->vector3 (protobuf/default Gui$NodeDesc :size)))
-            (dynamic visible (g/constantly false)))
-  (property size types/Vec3 ; Just for presentation.
-            (value (g/fnk [manual-size size-mode texture-size]
-                          (if (= :size-mode-auto size-mode)
-                            (or texture-size manual-size)
-                            manual-size)))
-            (set (fn [_evaluation-context self _old-value new-value]
-                   (g/set-property self :manual-size new-value)))
-            (dynamic read-only? (g/fnk [size-mode] (= :size-mode-auto size-mode))))
+  (property manual-size types/Vec3 (default (protobuf/vector4->vector3 (protobuf/default Gui$NodeDesc :size))))
+  (property texture-size types/Vec3 ; Just for presentation.
+            (value (g/fnk [anim-data]
+                     (if (some? anim-data)
+                       [(float (:width anim-data)) (float (:height anim-data)) protobuf/float-zero]
+                       protobuf/vector3-zero)))
+            (dynamic read-only? (g/constantly true))
+            (dynamic visible (g/fnk [texture]
+                               (coll/not-empty texture))))
```

Comment:
Now we always show the Size property, and we label it "Manual Size". If there is a texture assigned, we also show the "Texture Size" property next to it. Even if a texture is applied, the user can edit "Manual Size" if they desire. This makes it more clear what is going on when overriding these properties.

Reply:
Taking into account that in runtime node has `size` this change makes it less clear.

I read code, and the whole setup looks tricky - yes =/